### ISSUE TITLE
Support for "loading" in combo boxes

### ISSUE THREAD

**Issue by dcefram** [salesforce/design-system-react#2235]
Hi, I'd like to first confirm that the current combobox does not yet support a loading state? Something like this in the lightningdesignsystem component blueprint: https://lightningdesignsystem.com/components/combobox/#User-input-(Typeahead)-Loading

If my understanding is correct that this is not yet supported, I'd like to know if there are plans to support this for search fields that would fetch data asynchronously. I'd rather not reinvent the wheel if this is already supported and is just somewhat not documented (or I'm just blind that I missed it in the docs), but if it is true that this is not implemented, then I'll have to make my own combobox based on the blueprint.

P.S. I was supposed to post this in SO, but I'm afraid that it won't be seen as there are no other posts there with the tag `design-system-react`.

Thanks!

**Comment by interactivellama** [comment]
No, issue with posting here. This is a roadmap question. You are correct, there is currently no loading state for Combobox. It hasn't been requested by a product team that the maintainers of this project are aligned with. Free free to contribute it though!

That said, you might be able to hack it with a custom menu item via `onRenderMenuItem`, maybe you could render one item with the spinner in it and some spacing than then replace it with the real data when it loads that way the rest the functionality is there for you.
